packages/form/src/hooks/useFormState.ts

```diff
@@ -0,0 +1,29 @@
+import type { FieldValues } from 'react-hook-form'
+import { useFormContext } from 'react-hook-form'
+
+type UseFormStateParams = {
+  subscription?: Record<string, boolean>
+}
+
+/**
+ * @deprecated
+ */
+export const useFormStateDeprecated = <TFieldValues extends FieldValues>(
```

Comment:
We cant have useXDeprecated everywhere it should be retrocompatible with actual useFormState, yes I know it can be redundant but the migration will be easier